### PR TITLE
[Feature] Text Units

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Description

This PR activates the text-unit feature in Artemis. TextUnits can be written in Markdown (image upload also supported) and will be rendered for the student as HTML. This feature allows instructors to move some of their lecture content from the traditional lecture slides as pdf fully into Artemis

Additional Features:
- Changes to the Markdown get saved to the local storage so instructors do not lose their progress
- There is a "open as popup" feature that opens the text unit content into a separate page. This allows for easy printing and printing to PDF for students who like to have everything as PDF.

### Steps for Testing

Follow the videos described in the screenshot section and just test that everything works. Create, Update, Delete...

### Test Coverage

Server:
image
Frontend:
image
image
image

### Screenshots

Create Video: https://streamable.com/ujo76r
Cache Video: https://streamable.com/m7ryv8
Student View Video: https://streamable.com/5knqvk

### CODE REVIEW

**Review by derLalla — CHANGES_REQUESTED** [review]
image

When clicking on `View in Popup` the resulting window is very small, you could try to give it a greater width and height on startup. 

Besides that this is really well done! I have tested all the other cases.

**Comment by stefanwaldhauser** [comment]
> image
> 
> When clicking on `View in Popup` the resulting window is very small, you could try to give it a greater width and height on startup.
> 
> Besides that this is really well done! I have tested all the other cases.

Done in ee9efabdb8751e420a62040ef13a8a0d3ce43320

**Review by tiffca — DISMISSED** [review]
Tested the test cases on test server 1 and all works well, nice changes! 

Tested creation, the cache, and viewing by student. Used sample markdown from the internet.

**Review by derLalla — DISMISSED** [review]
the window size is better now. Approved.

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/textUnit.service.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from 'app/app.constants';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { TextUnit } from 'app/entities/lecture-unit/textUnit.model';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { LectureUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/lectureUnit.service';
+
+type EntityResponseType = HttpResponse<TextUnit>;
+type EntityArrayResponseType = HttpResponse<TextUnit[]>;
```
Comment (FrankeLukas) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/textUnit.service.ts:10]: GitHub Actions complains: `'EntityArrayResponseType' is defined but never used`

**Review by jpbernius — CHANGES_REQUESTED** [review]
Can you check the CSS for tables? Example Markdown:
```
| Column 1       | Column 2     | Column 3     |
| :------------- | :----------: | -----------: |
|  Cell Contents | More Stuff   | And Again    |
| You Can Also   | Put Pipes In | Like this \| |
```

No borders are rendered in the box.
image
It is better for the popup:
image

File: src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        $event.stopPropagation();
+
+        const width = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+        const height = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
+        const win = window.open('', ' ', 'toolbar=no,location=no,directories=no,status=no,menubar=no,scrollbars=yes,resizable=yes,' + `width=${width},height=${height}`);
+        win!.document.write(`<html><head><title>${this.textUnit.name}</title>`);
+        win!.document.write(
+            '<link rel="stylesheet" ' +
+                'href="https://cdnjs.cloudflare.com/ajax/libs/github-markdown-css/4.0.0/github-markdown.min.css" ' +
+                'integrity="sha512-Oy18vBnbSJkXTndr2n6lDMO5NN31UljR8e/ICzVPrGpSud4Gkckb8yUpqhKuUNoE+o9gAb4O/rAxxw1ojyUVzg==" crossorigin="anonymous" />',
```
Comment (jpbernius) [inline src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts:41]: Please don't embed external CS like that.

**Review by jpbernius — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<div *ngIf="isLoading" class="d-flex justify-content-center">
+    <div class="spinner-border" role="status">
+        <span class="sr-only">{{ 'loading' | translate }}</span>
+    </div>
+</div>
+
+<div *ngIf="!isLoading" class="container">
+    <jhi-text-unit-form [isEditMode]="false" (formSubmitted)="createTextUnit($event)"></jhi-text-unit-form>
```
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html:8]: Can you be more specific? What exactly should be refactored?

File: src/main/webapp/i18n/de/lectureUnit.json [hunk 4]
```diff
@@ -13,6 +14,31 @@
...
+                "content": "Inhalt",
+                "namePlaceHolder": "Gib der Texteinheit einen Namen",
+                "nameRequiredValidationError": "Der Name ist ein Pflichtfeld.",
+                "nameMaxLengthValidationError": "Der Name kann maximal 255 Zeichen lang sein."
+            },
+            "editTextUnit": {
+                "title": "Bearbeite Texteinheit"
+            },
+            "delete": {
+                "question": "Willst du wirklich die Texteinheit <strong>{{ title }}</strong> löschen?Du kannst diese Aktion nicht rückgängig machen!",
```
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/i18n/de/lectureUnit.json:37]: Done in e881a30d6467f0e45c64b9490f1b0b6a4fef22f0

File: src/main/webapp/i18n/de/lectureUnit.json [hunk 5]
```diff
@@ -13,6 +14,31 @@
             },
             "releaseDate": "Veröffentlichungsdatum"
         },
+        "textUnit": {
+            "created": "Eine neue Texteinheit erstellt",
+            "updated": "Texteinheit geupdated",
+            "tooltip": "Das ist eine Texteinheit.",
+            "popup": "In Popup öffnen",
+            "notReleasedTooltip": "Text nur sichtbar für Tutoren und Dozenten. Veröffentlichkeitsdatum:",
+            "markdownHelp": "Hier kann Markdown eingegeben werden. Mehr informationen: ",
```
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/i18n/de/lectureUnit.json:23]: Done in e881a30d6467f0e45c64b9490f1b0b6a4fef22f0

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/textUnit.service.ts (hunk 1 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/textUnit.service.ts:10]: Done in e881a30d6467f0e45c64b9490f1b0b6a4fef22f0

File: src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts (hunk 2 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts:41]: What is the preferred way to do it?

**Comment by stefanwaldhauser** [comment]
> Can you check the CSS for tables? Example Markdown:
> 
> ```
> | Column 1       | Column 2     | Column 3     |
> | :------------- | :----------: | -----------: |
> |  Cell Contents | More Stuff   | And Again    |
> | You Can Also   | Put Pipes In | Like this \| |
> ```
> 
> No borders are rendered in the box.
> image
> It is better for the popup:
> image

This is broken basically everywhere where we use markdown. 
image

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-text-unit/edit-text-unit.component.html [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div *ngIf="isLoading" class="d-flex justify-content-center">
+    <div class="spinner-border" role="status">
+        <span class="sr-only">{{ 'loading' | translate }}</span>
+    </div>
+</div>
+
+<div *ngIf="!isLoading" class="container">
+    <jhi-text-unit-form [formData]="formData" (formSubmitted)="updateTextUnit($event)" [isEditMode]="true"></jhi-text-unit-form>
```
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-text-unit/edit-text-unit.component.html:8]: Can you be more specific? What exactly should be refactored?

**Comment by stefanwaldhauser** [comment]
> Can you check the CSS for tables? Example Markdown:
> 
> ```
> | Column 1       | Column 2     | Column 3     |
> | :------------- | :----------: | -----------: |
> |  Cell Contents | More Stuff   | And Again    |
> | You Can Also   | Put Pipes In | Like this \| |
> ```
> 
> No borders are rendered in the box.
> image
> It is better for the popup:
> image

Fixed it globally in c60c2778058d5dfbe610fb8c4e40feefe19d3ff6. Now works on all markdown fields we use. The problem was that the table styling in bootstrap it opt-in, meaning the `table` class has to be added to a table for styling. I modified the markdown service to always add this class to created table tags
image

File: src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts (hunk 2 above)
Comment (jpbernius) [reply src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts:41]: Best way would be to serve it from /public I guess?

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpbernius) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html:8]: I'd suggest creating a `<lecture-unit-layout/>` component with something like this as its template:
```html
<div *ngIf="isLoading" class="d-flex justify-content-center">
     <div class="spinner-border" role="status">
         <span class="sr-only">{{ 'loading' | translate }}</span>
     </div>
 </div>

 <div *ngIf="!isLoading" class="container">
     <ng-content></ng-content>
</div>
```

To be used here like so:
```html
<lecture-unit-layout>
    <jhi-text-unit-form [isEditMode]="false" (formSubmitted)="createTextUnit($event)"></jhi-text-unit-form>
</lecture-unit-layout>
```

That way the spinner and the layout would be reused for all unit types.

**Review by FuchsDominik — APPROVED** [review]
So test wise everything seems great. Tested the features of text units and didnt experience any wrong behavior. Also the cache feature is nice.

File: src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts (hunk 2 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/overview/course-lectures/text-unit/text-unit.component.ts]: Thanks for the tip @jpbernius :) 
Done in: e2d2b5836e4fe89f36926dd66c29548072587c42

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-text-unit/create-text-unit.component.html:2]: Thanks for the tip. Done in 74163fbd896bbe491df078297b2024f5928bd1d4

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-text-unit/edit-text-unit.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-text-unit/edit-text-unit.component.html:2]: Done in 74163fbd896bbe491df078297b2024f5928bd1d4

**Review by tiffca — DISMISSED** [review]
Tested successfully on test server 3. The text units with markdown work as expected, including the cache feature. Nice change!

**Review by FuchsDominik — DISMISSED** [review]
Re-approve

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    let textUnitFormComponentFixture: ComponentFixture<TextUnitFormComponent>;
+    let textUnitFormComponent: TextUnitFormComponent;
+    let getItemStub: SinonStub;
+    let removeItemStub: SinonStub;
+    let setItemStub: SinonStub;
+    beforeEach(() => {
+        // mocking router
+        // mocking the local storage for cache testing
+        store = {};
+        getItemStub = sandbox.stub(localStorage, 'getItem').callsFake((key: string) => {
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts:39]: GitHub Actions: `'getItemStub' is assigned a value but never used`

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+            return store[key] || null;
+        });
+        removeItemStub = sandbox.stub(localStorage, 'removeItem').callsFake((key: string) => {
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts:42]: GitHub Actions: `'removeItemStub' is assigned a value but never used`

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+            delete store[key];
+        });
+        setItemStub = sandbox.stub(localStorage, 'setItem').callsFake((key: string, value: string) => {
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts:45]: GitHub Actions: `'setItemStub' is assigned a value but never used`

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+                TextUnitComponent,
+                MockComponent(FaIconComponent),
+                MockPipe(TranslatePipe),
+                MockDirective(NgbCollapse),
+                MockDirective(NgbTooltip),
+                MockPipe(ArtemisDatePipe),
+            ],
+            providers: [
+                MockProvider(ArtemisMarkdownService, {
+                    safeHtmlForMarkdown: (markdown: string) => exampleHTML,
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:42]: GitHub Actions: `'markdown' is defined but never used`

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+                    htmlForMarkdown: (markdown: string) => exampleHTML,
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:43]: GitHub Actions: `'markdown' is defined but never used`

**Review by FrankeLukas — APPROVED** [review]
Followed the videos. Works as advertised.

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts (hunk 11 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:43]: i mean i need to have the parameter to mock the method? Any idea how to do this? @FrankeLukas

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts (hunk 10 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:42]: i mean i need to have the parameter to mock the method? Any idea how to do this? @FrankeLukas

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts (hunk 9 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts]: done in d67c60cb6977715928970ad04df3a2f288f31f6b

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts (hunk 8 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts]: d67c60cb6977715928970ad04df3a2f288f31f6b

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts (hunk 7 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit-form.component.spec.ts]: d67c60cb6977715928970ad04df3a2f288f31f6b

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts (hunk 10 above)
Comment (FrankeLukas) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:42]: I think it wants something like `safeHtmlForMarkdown: () => exampleHTML,` here.

Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:42]: Yes but I can not do it as the mocked method has to match the signature of the original method

Comment (FrankeLukas) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:42]: Then I don't know. One could probably define a dummy method, but adding some sort of ignore annotation would be better if that is possible? (Unfortunately I don't know if that exists or if the errors here even matter after the PR is merged.)

**Review by derLalla — DISMISSED** [review]
re-approve

File: src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts (hunk 10 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:43]: 74049e8c98e6d469300d1d71e63d236bdb4a5d4d

Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/text-unit.component.spec.ts:43]: 74049e8c98e6d469300d1d71e63d236bdb4a5d4d

**Review by sjagla27 — APPROVED** [review]
Looks good and works great. Good job :)

**Review by FuchsDominik — APPROVED** [review]
re-approve

**Review by derLalla — DISMISSED** [review]
re-approve

**Review by nutatuna — DISMISSED** [review]
Tested on my machine and it worked.

**Review by schultek — DISMISSED** [review]
Creation, Editing with Cache and Deleting Text Units worked fine. For me it opened another tab instead of a popup for printing, but this will be because of my browser settings. Printing worked fine anyways. All in all very cool feature. 👍

**Review by schultek — APPROVED** [review]
Re-approve.

**Review by FuchsDominik — APPROVED** [review]
re-approve

File: src/main/resources/public/content/github-markdown.css [hunk 12]
```diff
@@ -1,985 +1,735 @@
 .markdown-body .octicon {
-  display: inline-block;
-  fill: currentColor;
-  vertical-align: text-bottom;
+    display: inline-block;
+    fill: currentColor;
+    vertical-align: text-bottom
 }
```
Comment (derLalla) [inline src/main/resources/public/content/github-markdown.css:6]: is this now an autogenerated file? Because a lot of line were deleted

**Review by derLalla — APPROVED** [review]
re-approve

File: src/main/resources/public/content/github-markdown.css (hunk 12 above)
Comment (stefanwaldhauser) [reply src/main/resources/public/content/github-markdown.css:6]: I ran it through a css cleaner

**Review by nutatuna — APPROVED** [review]
Tested the new version. Re-approve.

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'jhi-lecture-unit-layout',
+    templateUrl: './lecture-unit-layout.component.html',
+    styles: [],
+})
+export class LectureUnitLayoutComponent implements OnInit {
```
Comment (jpbernius) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts:8]:
```suggestion
export class LectureUnitLayoutComponent {
```

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import { Component, Input, OnInit } from '@angular/core';
```
Comment (jpbernius) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts:1]:
```suggestion
import { Component, Input } from '@angular/core';
```

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    templateUrl: './lecture-unit-layout.component.html',
+    styles: [],
+})
+export class LectureUnitLayoutComponent implements OnInit {
+    @Input()
+    isLoading = false;
+
+    constructor() {}
+
+    ngOnInit(): void {}
```
Comment (jpbernius) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts:14]:
```suggestion
```

File: src/test/javascript/spec/component/lecture-unit/text-unit/create-text-unit.component.spec.ts [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+@Component({ selector: 'jhi-text-unit-form', template: '' })
+class TextUnitFormStubComponent {
+    @Input() isEditMode = false;
+    @Output() formSubmitted: EventEmitter<TextUnitFormData> = new EventEmitter<TextUnitFormData>();
+}
+
+@Component({ selector: 'jhi-lecture-unit-layout', template: '<ng-content></ng-content>' })
+class LectureUnitLayoutStubComponent {
+    @Input()
+    isLoading = false;
```
Comment (jpbernius) [inline src/test/javascript/spec/component/lecture-unit/text-unit/create-text-unit.component.spec.ts:31]:
```suggestion
    @Input() isLoading = false;
```

File: src/test/javascript/spec/component/lecture-unit/text-unit/edit-text-unit.component.spec.ts [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+class TextUnitFormStubComponent {
+    @Input() isEditMode = false;
+    @Input() formData: TextUnitFormData;
+    @Output() formSubmitted: EventEmitter<TextUnitFormData> = new EventEmitter<TextUnitFormData>();
+}
+
+@Component({ selector: 'jhi-lecture-unit-layout', template: '<ng-content></ng-content>' })
+class LectureUnitLayoutStubComponent {
+    @Input()
+    isLoading = false;
```
Comment (jpbernius) [inline src/test/javascript/spec/component/lecture-unit/text-unit/edit-text-unit.component.spec.ts:33]:
```suggestion
    @Input() isLoading = false;
```

**Review by jpbernius — COMMENTED** [review]
Some nitpicking:

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts (hunk 13 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts (hunk 14 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts (hunk 15 above)
Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

Comment (stefanwaldhauser) [reply src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-layout/lecture-unit-layout.component.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

File: src/test/javascript/spec/component/lecture-unit/text-unit/edit-text-unit.component.spec.ts (hunk 17 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/edit-text-unit.component.spec.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

File: src/test/javascript/spec/component/lecture-unit/text-unit/create-text-unit.component.spec.ts (hunk 16 above)
Comment (stefanwaldhauser) [reply src/test/javascript/spec/component/lecture-unit/text-unit/create-text-unit.component.spec.ts]: f8b3a8c32952d73bebf274e32e506db2f599a1f6

File: src/main/resources/public/content/github-markdown.css [hunk 18]
```diff
@@ -0,0 +1,735 @@
+.markdown-body .octicon {
+    display: inline-block;
+    fill: currentColor;
+    vertical-align: text-bottom
```
[the remark this answers is not in the document]
Comment (stefanwaldhauser) [reply src/main/resources/public/content/github-markdown.css]: done :)

**Review by FuchsDominik — APPROVED** [review]
Re-re-approve

**Review by jpbernius — APPROVED** [review]
Awesome! LGTM! :shipit:

**Pull request merged**